gulpfile.js

```diff
@@ -14,31 +14,21 @@ const glob = require('glob');
 
 const env = process.env;
 
-gulp.task('webpack-dev', (cb) => {
+gulp.task('webpack-dev', async () => {
     preWebpack();
-    spawn(path.join(__dirname, './node_modules/.bin/webpack'), ['--mode', 'development'], { stdio: 'inherit', env }, cb);
+    return await spawn(path.join(__dirname, './node_modules/.bin/webpack'), ['--mode', 'development'], { stdio: 'inherit', env });
 });
 
-gulp.task('webpack-prod', (cb) => {
+gulp.task('webpack-prod', async () => {
     preWebpack();
-    spawn(path.join(__dirname, './node_modules/.bin/webpack'), ['--mode', 'production'], { stdio: 'inherit', env }, cb);
-});
-
-gulp.task('test', ['install-azure-account'], (cb) => {
-    env.DEBUGTELEMETRY = 1;
-    env.CODE_TESTS_WORKSPACE = path.join(__dirname, 'test/test.code-workspace');
-    env.CODE_TESTS_PATH = path.join(__dirname, 'dist/test');
-    const cmd = cp.spawn('node', ['./node_modules/vscode/bin/test'], { stdio: 'inherit', env });
-    cmd.on('close', (code) => {
-        cb(code);
-    });
+    return await spawn(path.join(__dirname, './node_modules/.bin/webpack'), ['--mode', 'production'], { stdio: 'inherit', env });
 });
 
 /**
  * Installs the azure account extension before running tests (otherwise our extension would fail to activate)
  * NOTE: The version isn't super important since we don't actually use the account extension in tests
  */
-gulp.task('install-azure-account', () => {
+gulp.task('install-azure-account', async () => {
```

Comment:
Why the async?

Reply:
I think I'd prefer to keep all tasks async or all non-async.  With async, I don't have to worry about returning Promise.resolve(), and I can use await.

Reply:
Actually I think @PrashanthCorp was right to question this. You _can_ return a Promise to signal completion, but [the value is ignored](https://github.com/gulpjs/gulp/blob/master/docs/getting-started/4-async-completion.md#returning-a-promise). Thus when we returned a `Stream` or `ChildProcess` it was ignored. I used [this branch](https://github.com/Microsoft/vscode-docker/commits/ej/test2) to test the theory (The commit `Test` broke the tests, but passed CI. The commit `Remove async` addressed the async thing and properly reported a failure)
    ![screen shot 2018-12-06 at 11 49 41 am](https://user-images.githubusercontent.com/11282622/49608270-0f57a400-f94d-11e8-8b6e-f2bd8274294e.png)

Anyways I'll fix this in an upcoming PR